Functional testing instructions for Magento Cloud code contributions (#7019)
magento/devdocs: src/cloud/docker/docker-test-magecloud-pkg-code.md

   ```

1. Add required project dependencies if you are testing the following packages.

   -  For the `{{site.data.var.mcp-package}}` package:

      ```bash
      composer require "magento/magento-cloud-docker:^1.0.0" --no-update
      ```

   -  For the `{{site.data.var.mcc-package}}` package:

      ```bash
      composer require "magento/magento-cloud-docker:^1.0.0" --no-update
      composer require "magento/framework:*" --no-update
      composer require "magento/module-store:*" --no-update
      composer require "magento/module-url-rewrite:*" --no-update
      ```

1. Switch to the preferred PHP version for running tests.

1. Update the project dependencies.

   ```bash
   composer update
   ```

### Add credentials

Add credentials to your Docker environment using any of the following methods:

-  Use environment variables
-  Load credentials from the environment configuration file
-  Add variables directly to the test configuration file

{:.procedure}
To add credentials using environment variables:

1. Create the following environment variables:

   ```bash
   export REPO_USERNAME=your_public_key
   ```

   ```bash
   export REPO_PASSWORD=your_private_key
   ```

> 1. To install some packages from GitHub, create an environment variable for the GitHub token:

Author: ```suggestion
1. Some packages require a GitHub authentication during installation. Create an environment variable with your GitHub token that can be used to install these packages. 
```

> 1. Some packages require a GitHub authentication during installation. Create an environment variable with your GitHub token that can be used to install these packages. 

Author: ```suggestion
1. Some packages require a GitHub authentication during installation. Create an environment variable with your GitHub token that can be used to install these packages.
```

   ```bash
   `export GITHUB_TOKEN=your_github_token
   ````

{:.procedure}
To load credentials from the environment configuration file:

1. Run the following commands to write credentials to the ./.env file.

   ```bash
   echo "REPO_USERNAME=your_public_key" >> ./.env
   ```

   ```bash
   echo "REPO_PASSWORD=your_private_key" >> ./.env
   ```

1. Edit the `codeception.dist.yml` file:

   ```yaml
   params:
       - tests/functional/configuration.dist.yml
       - env
       - .env
   ```

1. Add a dependency for the [vlucas/phpdotenv] package required to load the environment variables.

   ```bash
   composer require "vlucas/phpdotenv": "^3.0"
   ```

{:.procedure}
To add credentials directly to the test configuration file:

> 1. Edit the `codeception.dist.yml` file.

Author: I made it a two step procedure to accomodate the  {:.procedure} class title so that this section would be formatted the same as the others.  Initially, these three procedures were all combined into a single procedure.

> 1. Edit the `codeception.dist.yml` file.

Author: ```suggestion
1. Open the `codeception.dist.yml` file in an editor.
```
